### PR TITLE
 936 Validation Factory 

### PR DESCRIPTION
**Pull request by wbyoko**
* Empty Test Cases https://github.com/flexion/ef-cms/blob/6e2d4d9621fd028d8da32003f8202a5c79ea2841/shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.test.js

* Build a validation object based off of metadata
* update tabs proptype

### CODE REVIEW

**Review by codyseibert — APPROVED** [review]

File: shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js [hunk 1]
```diff
@@ -14,11 +15,173 @@ function ExternalDocumentInformationFactory() {}
...
+  const addProperty = (itemName, itemSchema, itemErrorMessage) => {
+    schema[itemName] = itemSchema;
+    if (itemErrorMessage) {
+      errorToMessageMap[itemName] = itemErrorMessage;
+    }
+  };
+
+  const supportingDocumentFreeTextCategories = [
+    'Affidavit',
+    'Declaration',
```
Comment (rachaelparris) [inline shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js]: All of these actually need `in Support` at the end of them. The names without `in Support` are just for display.

File: shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js [hunk 2]
```diff
@@ -14,11 +15,173 @@ function ExternalDocumentInformationFactory() {}
...
+        'Motion to Withdraw Counsel',
+        'Motion to Withdraw As Counsel',
+        'Application to Take Deposition',
+      ],
+      documentMetadata.documentType,
+    )
+  ) {
+    // NOTE: objections isn't a boolean
+    addProperty('objections', joi.string().required(), [
+      'Objections is required.',
```
Comment (rachaelparris) [inline shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js:90]: Should it be a boolean? Is the plan to store the `Yes`, `No`, or `Unknown` as a string?

**Review by kkoskelin — APPROVED** [review]
(approved once Rachael's comments are addressed)

File: shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js (hunk 2 above)
Comment (wbyoko) [reply shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js:90]: I'm not sure. Were we going to ignore `Unknown`?

Comment (rachaelparris) [reply shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js:90]: Yeah, I'm thinking it should be a string. Looks good!

**Review by rachaelparris — APPROVED** [review]

**Pull request merged**